dashboard/modules/state/state_head.py

```diff
@@ -35,14 +41,78 @@ class StateHead(dashboard_utils.DashboardHeadModule):
     ray.list_actors(), ray.get_actor(), ray.summary_actors().
     """
 
-    def __init__(self, dashboard_head):
+    def __init__(
+        self,
+        dashboard_head,
+    ):
+        """Initialize for handling RESTful requests from State API Client"""
         super().__init__(dashboard_head)
         self._state_api_data_source_client = None
         self._state_api = None
         self._log_api = None
+
+        # Rate limiting related fields
+        self._num_requests_in_progress = 0
+        self._max_http_req_in_progress = RAY_STATE_SERVER_MAX_HTTP_REQUEST
+        if self._max_http_req_in_progress > RAY_STATE_SERVER_MAX_HTTP_REQUEST_ALLOWED:
+            # We don't allow users to configure too high a rate limit
+            self._max_http_req_in_progress = RAY_STATE_SERVER_MAX_HTTP_REQUEST_ALLOWED
+
         DataSource.nodes.signal.append(self._update_raylet_stubs)
         DataSource.agents.signal.append(self._update_agent_stubs)
 
+    def enforce_max_concurrent_calls(func):
```

Comment:
Is there an easy way to make this method generally useable across the API server? (ooc)

Reply:
Good idea - refactored.